catroid/src/main/java/org/catrobat/catroid/ui/recyclerview/fragment/ProjectListFragment.java

```diff
@@ -388,9 +388,13 @@ public void onItemClick(ProjectData item) {
 		}
 		if (actionModeType == NONE) {
 			setShowProgressBar(true);
-			new ProjectLoadTask(item.getDirectory(), getContext())
-					.setListener(this)
-					.execute();
+			if (item.getDirectory() != null) {
```

Comment:
showing the error is already implemented, see `onLoadFinished()`, please move checking the directory against nullability inside `ProjectLoadTask`. also please provide some tests for that.

Reply:
Done, also added corresponding test to check nullability. If this doesn't solve the issue the only way to handle this is to check nullability in  `loadProject()`  of `XstreamSerializer` , which is highly unlikely to give error as there are already some exception checks.